LINK-85 | Recurring events

jorilindell opened the pull request:
## Description
Modal component
Modal to enter recurring event settings
Collapsible component
List of recurring events and possibility to delete them

## Closes
[LINK-85](https://helsinkisolutionoffice.atlassian.net/browse/LINK-85)

#Screenshots
<img width="1164" alt="Screenshot 2020-11-23 at 14 01 51" src="https://user-images.githubusercontent.com/24706814/99960493-7f0f0580-2d95-11eb-83df-2d522c8aa2d5.png">


<img width="699" alt="Screenshot 2020-11-23 at 14 03 03" src="https://user-images.githubusercontent.com/24706814/99960495-82a28c80-2d95-11eb-9304-4ae70dc22de4.png">

Sartonon commented on src/domain/event/formSections/timeSection/TimeSection.tsx:
```diff
@@ -1,20 +1,58 @@
+  };
+
+  const addRecurringEventSettings = (values: RecurringEventSettings) => {
+    recurringEvents.push(values);
+    setRecurringEvents(recurringEvents);
```
Could this be
`setRecurringEvents([...recurringEvents, values]);`
to avoid mutation

jorilindell replied:
Oh yes, changed

Sartonon commented on src/assets/theme/theme.ts:
```diff
@@ -127,6 +127,19 @@ const theme: Theme = {
     '--label-color': 'var(--color-black-90)',
     '--label-color-disabled': 'var(--color-black-40)',
   },
+  collapsible: {
+    '--heading-background-color': 'var(--color-white)',
+    '--heading-border-color': 'var(--color-bus)',
+    '--heading-color': 'var(--color-bus)',
+    '--heading-background-color-expanded': 'var(--color-bus)',
+    '--heading-border-color-expanded': 'var(--color-bus)',
+    '--heading-color-expanded': 'var(--color-white)',
+    '--heading-background-color-expanded-hover': 'var(--color-bus-dark)',
+    '--heading-border-color-expanded-hover': 'var(--color-bus-dark)',
+    '--heading-color-expanded-hover': 'var(--color-white)',
+    '--heading-background-color-hover': 'var(--color-bus-light)',
+    '--heading-color-hover': 'var(--color-bus-dark)',
+  },
```
btw do these variables override the ones in collapsible.module.scss?

jorilindell replied:
Yes they do. The idea is to have the all theme css variables in one single file

Sartonon reviewed (commented):
Couple comments, will continue review later to tomorrow

Sartonon commented:
<img width="1035" alt="Screenshot 2020-11-24 at 17 23 44" src="https://user-images.githubusercontent.com/15219142/100115082-c622f680-2e7a-11eb-8044-3263a4a7fa52.png">

I get this error

jorilindell commented:
> <img alt="Screenshot 2020-11-24 at 17 23 44" width="1035" src="https://user-images.githubusercontent.com/15219142/100115082-c622f680-2e7a-11eb-8044-3263a4a7fa52.png">
> 
> I get this error

Fixed that. Changed it also so that Timepicker replaces : with . when onBlur is called

Sartonon commented:
<img width="1241" alt="Screenshot 2020-11-25 at 10 00 38" src="https://user-images.githubusercontent.com/15219142/100198861-19418b80-2f05-11eb-94d9-e7c99e625ace.png">

should this one have some top and bottom margin? :)

Sartonon reviewed (approved):
Very good!
